File: C/27. Remove Element.c
int removeElement(int* nums, int numsSize, int val){
    int r = 0;
    for (int i = 0; i < numsSize; i++){
        if (nums[i] == val){
            r++;
        }
        else{
            nums[i-r] = nums[i];
        }
    }
    return numsSize - r;
}

File: JavaScript/27. Remove Element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var r = 0;
    for (var i = 0; i < nums.length; i++){
        if (nums[i] == val){
            r++;
        }
        else{
            nums[i-r] = nums[i];
        }
    }
    return nums.length - r;
};
